src/dune_engine/package.ml

```diff
@@ -480,7 +480,7 @@ module Info = struct
       [ field_o "source" Source_kind.encode source
       ; field_l "authors" string (Option.value ~default:[] authors)
       ; field_l "maintainers" string (Option.value ~default:[] maintainers)
-      ; field_o "license" string license
+      ; field_l "license" string (Option.value ~default:[] license)
```

Comment:
What does this do in the presence of say `(license LGPL-2.1-or-later WITH OCaml-LGPL-linking-exeption)` ?

Reply:
It'll yield `license: ["LGPL-2.1-or-later" "WITH" "OCaml-LGPL-linking-exeption"]`.

Is `WITH` currently handled somewhere?

Reply:
in opam yes, this is very common. It should be outputed as one string as before

Reply:
Did you mean `(license "LGPL-2.1-or-later WITH OCaml-LGPL-linking-exeption")`?
Space-separated list was rejected before 3.2.

Reply:
ah i see, yes. Then it should be fine.

Reply:
An alternative to this would  be to introduce a `licenses` stanza instead. This similar as to how we have `executable` and `executables` stanzas for example, and it would avoid this gotcha.

@kit-ty-kate @liyishuai what do you think?

Reply:
i think it’s fine with just one as it reflects the opam field of the same name

Reply:
Okay, but isn't this going to break compat for people who have:

```
(lang dune 3.1)
(license LGPL-2.1-or-later WITH OCaml-LGPL-linking-exeption)
```

And they will upgrade to `3.2`?

Reply:
no, as @liyishuai said this dune-project file would result in an error:
```
$ cat dune-project
(lang dune 3.1)
(license LGPL-2.1-or-later WITH OCaml-LGPL-linking-exeption)
$ dune build
File "dune-project", line 2, characters 27-31:
2 | (license LGPL-2.1-or-later WITH OCaml-LGPL-linking-exeption)
                               ^^^^
Error: Too many argument for license
```